csc302-spring-2021/proj-DaTeam
# Shared tests

## Description

Implemented the testing framework for the shared module.

- Closes #90
- Implemented the existing tests (from Tests.ts) to jest tests. It replaces it so I removed the old testing file
- Split up the validation and serialization tests into two different test suites
- Updated the npm test scripts to include the new shared module's tests. Also moved the frontend tests last since for some reason when "No tests found related to files changed since last commit" it hangs (thus not running the backend or shared tests). 

## Checklist

- ~[ ] I have written unit tests for the code I added~ This change is for unit tests

## QA Steps

1. ```cd shared/```
1. ```npm test```
1. Verify 2 test suites ran, and all 34 tests pass ✔️ / ❌ 
2.  ```cd .. ```
3. ```npm test```
4. Verify that the shared, backend, and front end tests all still run as expected  ✔️ / ❌
